recipe/meta.yaml

```
@@ -16,26 +16,25 @@ build:
 
 requirements:
   build:
+    - {{ stdlib("c") }}
     - {{ compiler('cxx') }}
     - cmake
     - make
-  run_constrained:
-    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx]
+  host:
+    - zlib
+    - bzip2
 
 test:
   source_files:
     - source/test
   requires:
+    - {{ stdlib("c") }}
     - {{ compiler('cxx') }}
     - {{ compiler('fortran') }}
   commands:
-    # Workaround for lack of CONDA_BUILD_SYSROOT variable in staged recipes
-    - export CONDA_BUILD_SYSROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk  # [osx]
     - cd source/test
```

```suggestion
    # Check HepMC3-config values
    - HepMC3-config --prefix
    - HepMC3-config --cxxflags
    - HepMC3-config --ldflags
    - HepMC3-config --libdir

    - cd source/test
```

@Kwaizer can we see the output of these commands in the logs? This might be helpful to see why only macOS is failing here.

---

@matthewfeickert I can only see this in logs on azure for these commands: 
```
2024-08-22T20:11:36.6216790Z + HepMC3-config --prefix
2024-08-22T20:11:36.6227840Z /Users/runner/miniforge3/conda-bld/hepmc3_1724357047592/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/bin/HepMC3-config: line 8: greadlink: command not found
```

---

Ah interesting! That means that the version of `homebrew` that they have installed doesn't have `coreutils` added (c.f. https://formulae.brew.sh/formula/coreutils).

Hm. I don't know how to get Homebrew to update on conda-forge unless you are supposed to explicitly call that (but that seems bad). Though there is https://github.com/conda-forge/coreutils-feedstock so maybe we can try adding that to the `build` requirements for `  # [osx]`.